src/main/java/com/jcabi/http/wire/LastModifiedCachingWire.java

```diff
@@ -0,0 +1,211 @@
+                req, home, method, headers, content, connect, read
+            );
+        } else {
+            rsp = this.origin.send(
+                req, home, method, headers, content, connect, read
+            );
+        }
+        return rsp;
+    }
+
+    /**
+     * Check cache and update if needed.
+     *
+     * @param req Request
+     * @param home URI to fetch
+     * @param method HTTP method
+     * @param headers Headers
+     * @param content HTTP body
+     * @param connect The connect timeout
+     * @param read The read timeout
+     * @return Response obtained
+     * @throws IOException if fails
+     * @checkstyle ParameterNumber (6 lines)
+     */
+    private Response consultCache(final Request req,
+        final String home,
+        final String method,
+        final Collection<Map.Entry<String, String>> headers,
+        final InputStream content,
+        final int connect,
+        final int read) throws IOException {
+        final Response rspReceived;
+        if (this.cache.containsKey(req)) {
+            rspReceived = this.validateCacheWithServer(
+                req, home, method, headers, content, connect, read
+            );
+        } else {
+            rspReceived = this.origin.send(
+                req, home, method, headers, content, connect, read
+            );
+            this.updateCache(req, rspReceived);
+        }
+        return rspReceived;
+    }
+
+    /**
+     * Check response and update cache if needed.
+     *  @todo Evict cache entry if If-Modified-Since request responded with
+     *  HTTP_OK code and no Last-Modified header.
+     *
+     * @param req Request
+     * @param home URI to fetch
+     * @param method HTTP method
+     * @param headers Headers
+     * @param content HTTP body
+     * @param connect The connect timeout
+     * @param read The read timeout
+     * @return Response obtained
+     * @throws IOException if fails
+     * @checkstyle ParameterNumber (8 lines)
+     */
+    private Response validateCacheWithServer(final Request req,
+        final String home,
+        final String method,
+        final Collection<Map.Entry<String, String>> headers,
+        final InputStream content,
+        final int connect,
+        final int read) throws IOException {
+        final Response rspCached = this.cache.get(req);
+        final Collection<Map.Entry<String, String>> hdrs = this.enrich(
+            headers, rspCached
+        );
+        Response rsp = this.origin.send(
+            req, home, method, hdrs, content, connect, read
+        );
+        if (rsp.status() == HttpURLConnection.HTTP_NOT_MODIFIED) {
+            rsp = rspCached;
+        } else {
+            this.updateCache(req, rsp);
+        }
+        return rsp;
+    }
+
+    /**
+     * Add response to cache.
+     * @param req The key of the request
+     * @param rsp The response to add
+     */
+    private void updateCache(final Request req, final Response rsp) {
```

Comment:
@piddubnyi see http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html